docs/modules/ROOT/pages/variables-and-params.adoc

```diff
@@ -0,0 +1,384 @@
+For example:
+
+```Cypher
+MATCH (m:Movie)
+RETURN m
+```
+
+The variable `m` refers to a node matched in the pattern `(:Movie)`. The node is then returned.
+
+One of the biggest challenges of writing dynamic Cypher queries is that the resulting variables in the query need to have unique names that correctly reference the values. To achieve this, Cypher Builder provides an abstraction of variables in the form of JavaScript references, removing the need to track variable names.
+
+=== Types
+
+Cypher builder distinguishes four types of variables:
+
+* `Node`: Holds a reference to a Node matched in a pattern.
+* `Relationship`: Holds a reference to a Relationship in a pattern.
+* `Variable`: Generic variable reference.
+* `Path`: A variable to hold a path.
+
+All variables are just virtual references to a Cypher variable that, will be translated to a unique name.
+
+=== Defining a variable
+
+An arbitrary Cypher variable can be defined with:
+
+```javascript
+const myVar = new Cypher.Variable();
+```
+
+This variable can then be used as an expression in any clause that supports variables, for instance:
+
+```javascript
+const myVar = new Cypher.Variable();
+const query = new Cypher.Return(myVar)
+```
+
+```cypher
+RETURN var0
+```
+
+The name `var0` is automatically generated when `.build()` is called to prevent name collision. If the variable is reused, the same name will be used in the generated Cypher, for instance:
+
+```javascript
+const myVar = new Cypher.Variable();
+const query = new Cypher.With(myVar).return(myVar);
+```
+
+```cypher
+WITH var0
+RETURN var0
+```
+
+Each instance you create of a variable will be associated with a different name:
+
+```javascript
+const withVar = new Cypher.Variable();
+const returnVar = new Cypher.Variable();
+const query = new Cypher.With(withVar).return(returnVar);
+```
+
+```cypher
+WITH var0
+RETURN var1
+```
+
+=== Nodes and relationships
+
+Nodes and relationships are variables holding some extra data regarding nodes and relationships to be used in xref:patterns.adoc[Patterns].
+For example:
+
+```javascript
+const movie = new Cypher.Node({labels: ["Movie"]});
+const relationship = new Cypher.Relationship({type: "ACTED_IN"});
+
+const pattern = new Cypher.Pattern(movie).related(relationship).to();
+const match = new Cypher.Match(pattern).return(movie);
+```
+
+```Cypher
+MATCH (this0:Movie)-[this1:ACTED_IN]->(this2)
+RETURN this0
+```
+
+In this example, the node variable `movie` is created and then used in the `MATCH` and `RETURN` clauses.
+
+In the translated Cypher, you can see how the node variable is translated to the Cypher variable `this0`, while the relationship is assigned to `this1`. These names are not defined in your JavaScript code, but are automatically generated when the query is built with `.build()`. The labels and type passed to `Node` and `Relationship` variables are used in the Pattern.
+
+The following parameters can be passed to Node:
```

Comment:
I wonder whether this should be presented as a list. Will there be more parameters in the future, possibly?

```suggestion
The following parameters can be passed to a node:
```

Reply:
In this case Node is the name of the class to be called, so maybe it would be more accurate to say

```suggestion
The following parameters can be passed to a `Node` class:
```